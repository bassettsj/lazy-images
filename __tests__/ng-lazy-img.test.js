import 'angular';
import 'angular-mocks';
import ngLazyImg from '../src/ng-lazy-img';
import lazyImg from '../src/lazy-img';

jest.mock('../src/lazy-img');

describe('Lazy Images Angular directive', () => {
  let $compile;
  let $rootScope;
  const src = 'https://example.com/logo.png';
  beforeEach(window.module(ngLazyImg.name));

  beforeEach(window.inject((_$rootScope_, _$compile_) => {
    $compile = _$compile_;
    $rootScope = _$rootScope_;
  }));

  it('should observe the lazy src attribute', () => {
    const el = $compile(`<img lazy-img lazy-src="${src}"/>`)($rootScope);
    expect(el).toMatchSnapshot();
    expect(lazyImg).not.toHaveBeenCalled();
    $rootScope.$apply();
    expect(lazyImg).toHaveBeenCalled();
  });

  it('should throw an error when it called on another element', () => {
    const fn = () => {
      $compile(`<p lazy-img lazy-src="${src}"/>`)($rootScope);
    };
    expect(fn).toThrowErrorMatchingSnapshot();
  });
});
