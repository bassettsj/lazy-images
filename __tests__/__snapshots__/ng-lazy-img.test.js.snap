// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lazy Images Angular directive should observe the lazy src attribute 1`] = `
Object {
  "0": <img
    class="ng-scope"
    lazy-img=""
    lazy-src="https://example.com/logo.png"
  />,
  "length": 1,
}
`;

exports[`Lazy Images Angular directive should throw an error when it called on another element 1`] = `"Must be used with an image element"`;
